tests/components/august/test_binary_sensor.py

```diff
@@ -23,16 +23,14 @@
     lock_one = await _mock_lock_from_fixture(
         hass, "get_lock.online_with_doorsense.json"
     )
-    lock_details = [lock_one]
-    await _create_august_with_devices(hass, lock_details)
+    await _create_august_with_devices(hass, [lock_one])
```

Comment:
I browsed the mock module and I think the argument order in the setup function is wrong for the pytest fixtures and decorated mocks.

The pytest fixtures should go last.

https://github.com/home-assistant/home-assistant/blob/e9a7b66df67426f00529d76cafedc6300dc03ee3/tests/components/august/mocks.py#L46-L48

Reply:
@MartinHjelmare I tried changing the ordering all the test fails.

Reply:
AFAICT since I'm passing in hass and api_instance the mocks are applied bottom up per https://docs.python.org/3/library/unittest.mock.html#quick-guide

Reply:
What does the log say for one example?

Reply:
Right, sorry, I was thinking this was a test function for some reason. 🤦‍♂️

Reply:
```
hass = <homeassistant.core.HomeAssistant object at 0x7f55040456a0>

    async def test_create_lock_with_linked_keypad(hass):
        """Test creation of a lock with a linked keypad that both have a battery."""
        lock_one = await _mock_lock_from_fixture(hass, "get_lock.doorsense_init.json")
>       await _create_august_with_devices(hass, [lock_one])

tests/components/august/test_sensor.py:51: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/components/august/mocks.py:143: in _create_august_with_devices
    return await _mock_setup_august_with_api_side_effects(hass, api_call_side_effects)
tests/components/august/mocks.py:181: in _mock_setup_august_with_api_side_effects
    return await _mock_setup_august(hass, api_instance)
/usr/lib/python3.8/unittest/mock.py:1359: in patched
    return await func(*newargs, **newkeywargs)
tests/components/august/mocks.py:54: in _mock_setup_august
    assert await async_setup_component(hass, DOMAIN, _mock_get_config())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

hass = <MagicMock name='authenticate' id='140002913532752'>, domain = 'august', config = {'august': {'login_method': 'email', 'password': 'mocked_password', 'username': 'mocked_username'}}

    async def async_setup_component(
        hass: core.HomeAssistant, domain: str, config: Dict
    ) -> bool:
        """Set up a component and all its dependencies.
    
        This method is a coroutine.
        """
        if domain in hass.config.components:
            return True
    
        setup_tasks = hass.data.setdefault(DATA_SETUP, {})
    
        if domain in setup_tasks:
            return await setup_tasks[domain]  # type: ignore
    
        task = setup_tasks[domain] = hass.async_create_task(
            _async_setup_component(hass, domain, config)
        )
    
>       return await task  # type: ignore
E       TypeError: object MagicMock can't be used in 'await' expression

homeassistant/setup.py:49: TypeError
------------------------------------------------
```